### PR TITLE
[CDF-21406] Remove common folder function code

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by ronpal — APPROVED** [review]

File: CHANGELOG.cdf-tk.md
Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: In `cdf-tk-dev.py` the following is used:
`common_function_code="./common_function_code",`

Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: `cognite_toolkit/cognite_modules/default.environments.yaml`

```
common_function_code: ./common_function_code
...
common_function_code: ./common_function_code
```

Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: In ´cognite_toolkit/README.md`:
```
## Configuration Files
...
- `common_function_code` - The path to where common code for functions can be found.
```

Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: In file `demo/config.demo.yaml`:

`common_function_code: ./common_function_code`

Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: In `pyproject.toml`:
```
[tool.mypy]
exclude = ["cognite_toolkit/cognite_modules",
            "cognite_toolkit/common_function_code",  <-- maybe remove?
            "cognite_toolkit/build*"]
```

Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: The test: 
`def mock_environments_yaml_file(module_path: Path, monkeypatch: MonkeyPatch) -> None:`

Contains:
`"common_function_code": Path(REPO_ROOT / "cognite_toolkit/common_function_code").as_posix(),`

Comment (haakonvt) [inline CHANGELOG.cdf-tk.md:1]: File `tests/tests_unit/test_cdf_tk/run_data/config.dev.yaml`:
contains
`common_function_code: ../../../../cognite_toolkit/common_function_code`

**Review by haakonvt — CHANGES_REQUESTED** [review]
See other comments regarding files still mentioning common functions code

**Review by haakonvt — APPROVED** [review]
L G T M

**Pull request merged**